contracts/coordinator/src/contract/query.rs

```diff
@@ -14,3 +21,47 @@ fn is_verifier_in_any_verifier_set(deps: Deps, verifier_address: &Addr) -> bool
 pub fn check_verifier_ready_to_unbond(deps: Deps, verifier_address: Addr) -> StdResult<bool> {
     Ok(!is_verifier_in_any_verifier_set(deps, &verifier_address))
 }
+
+fn get_provers_for_verifier(
+    deps: Deps,
+    verifier_address: Addr,
+) -> Result<HashSet<Addr>, ContractError> {
+    let provers = VERIFIER_PROVER_INDEXED_MAP
+        .idx
+        .by_verifier
+        .prefix(verifier_address)
+        .range(deps.storage, None, None, Order::Ascending)
+        .map(|result| result.map(|(_, record)| record.prover))
+        .try_collect()?;
+
+    Ok(provers)
+}
+
+pub fn verifier_details_with_provers(
```

Comment:
prefer to move public functions to the top of the file and the private functions they use below them